### ISSUE TITLE
Monal IM, openfire, HttpUploadComponent, XEP-0363: HTTP File Upload

### ISSUE THREAD

**Issue by ageekhere** [siacs/HttpUploadComponent#70]
Has anyone got Monal IM image sending (XEP-0363) working with HttpUploadComponent using openfire?

I have got HttpUploadComponent connected with openfire https but when I try to send an image from monal im I get a connection issue from monal but the logs from HttpUploadComponent just shows an attempt to send a file with no error. anurodhp from monal says I need a valid cert from let's encrypt but not really sure what files HttpUploadComponent needs (.pem files?).

Someone has said they have had success with prosody and mod_http_upload.

**Comment by andersruneson** [comment]
Does httpupload work in clients Conversations or gajim (with httpupload
plugin) ?

**Comment by ageekhere** [comment]
Clients are unable to send images from monal using httpupload plugin.

**Comment by andersruneson** [comment]
The bug could as well be in monal, can you please test another client to
verify if it works there?

**Comment by ageekhere** [comment]
Someone reported that it works fine with prosody and mod_http_upload https://github.com/anurodhp/Monal/issues/30

Not sure what other client to test that uses XEP-0363

**Comment by andersruneson** [comment]
See my reply before??
https://github.com/siacs/HttpUploadComponent/issues/70#issuecomment-224176773

**Comment by ageekhere** [comment]
Yes, clients Conversations from monal to monal or jitsi through openfire work fine. Do not know what gajim is. I am new to HttpUploadComponent and was recommended to use it for openfire for XEP-0363 file upload for IOS.

Thanks

**Comment by anurodhp** [comment]
the vast majority of connection issues with http uploads appear to be related to one of the following: 
1. monal requires HTTPS
2. iOS requires a valid certificate
3. iOS requires a certain set of ciphers (https://github.com/anurodhp/Monal/issues/35#issuecomment-217653369)

**Comment by ageekhere** [comment]
Hi anurodhp, have you got http uploads to work with monal using openfire and HttpUploadComponent?

**Comment by anurodhp** [comment]
i haven't tested it on open fire. it does work on other servers. I will test it.

**Comment by tahatauqeer** [comment]
@anurodhp did you try to add this component to openfire?

**Comment by SergiySW** [comment]
You have to set up https key and valid certificate in HttpUploadComponent. Use put and get URLs with https prefix.
Certificate must be trusted be iOS (it have to be issued by trusted CA, additionally you can import your own CA in iOS).
Keys: RSA at least 2048 bit (and no more than 4096) or ECDSA at least 256 bit. Certificate fingerprint: SHA-2 at least SHA-256.
Check your HttpUploadComponent for support both TLSv1.2 and any cipher from list [in official ATS documentation](https://developer.apple.com/library/content/documentation/General/Reference/InfoPlistKeyReference/Articles/CocoaKeys.html)
